.claude/hooks/pre_tool_use_protect_precommit.py

```
@@ -0,0 +1,189 @@
+    return False
+
+
+def check_config_bash_command(command):
+    """Check if a bash command might interfere with configurations."""
+    for pattern, reason in BLOCKED_CONFIG_COMMANDS:
+        if re.search(pattern, command, re.IGNORECASE):
+            return False, reason
+    return True, None
+
+
+def check_content_for_config_changes(content):
```

Ruff (0.12.2)

Missing return type annotation for public function `check_content_for_config_changes`

(ANN201)